Client/game_sa/CWorldSA.cpp

```diff
@@ -443,6 +443,31 @@ float CWorldSA::FindGroundZFor3DPosition(CVector* vecPosition)
     return fReturn;
 }
 
+float CWorldSA::FindRoofZFor3DCoord(CVector* pvecPosition, bool* pbOutResult)
+{
+    DEBUG_TRACE("FLOAT CWorldSA::FindRoofZFor3DCoord(CVector * vecPosition, bool * pbOutResult)");
+
+    DWORD dwFunc  = FUNC_FindRoofZFor3DCoord;
+    FLOAT fReturn = 0;
+
+    FLOAT fX = pvecPosition->fX;
+    FLOAT fY = pvecPosition->fY;
+    FLOAT fZ = pvecPosition->fZ;
+
+    _asm
```

Comment:
In new code, we no longer allow asm blocks in methods that already have C++ code. Cast `FUNC_FindRoofZFor3DCoord` to a "function pointer" and then call that function.

Reply:
Yes.
```
auto CWorld_FindRoofZFor3DCoord = (float(__cdecl*)(CVector, bool*))0x0569750;
return CWorld_FindRoofZFor3DCoord(*pvecPosition, pbOutResult);
```

Reply:
:(
I was so happy I learned to copy paste ASM code
Sure, no problem btw. I'll do it.

Reply:
Why is it `__cdecl*`, instead of `__cdecl`? Im just curious

Reply:
Since `0x0569750` is the address of the function, we need to construct a function pointer and not the function itself. You can't put a function inside a variable, but a pointer works fine.

Reply:
Ah, that makes sense, thanks!